manager/runner.go

```diff
@@ -38,6 +38,10 @@ type Runner struct {
 	ErrCh  chan error
 	DoneCh chan struct{}
 
+	// ServerCh is a channel to surface error responses from the server up the calling stack
+	// and will only hold a maximum of one error at a time
+	ServerCh chan error
```

Comment:
This is a little bit of a nit, but `ServerCh` doesn't immediately indicate to me that this will contain errors, and it has me asking myself: what's the difference between `ErrCh` and `ServerCh`? Would something like `TemplateErrCh` be more clear? Maybe I'm not fully understanding the difference

Reply:
This is a good question. I wanted to try to distinguish between any template error (for example it could be a lookup error) and an error that is coming directly from the server it is making a request to. 
TemplateServerErr seemed weirder because it could imply that the template server had an issue (especially when it is being referenced in the Agent code). 

If you think one of these is better though I'm open to changing it.

Reply:
Actually - I'll change this to ServerErrCh !

Reply:
Hmm, that's a great point. I'm okay with keeping it as-is, but I still worry it could be confusing between the template server and the Vault server. Though I think any name we pick will probably be imperfect

Some more options to consider:
- `VaultRequestErrCh`
- `RequestErrCh`
- `TemplateRenderingErrCh`

I'm happy to keep it as is, like I've said, and naming things is hard, but if there's something clearer we can think of I think that'd be great. If you like any other option I'd say go for it, otherwise, it's fine as-is :D

Reply:
I like `ServerErrCh` more than `ServerCh`, thanks for humouring me!